org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/BundleUtilsTest.java

```diff
@@ -0,0 +1,102 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Microsoft Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Microsoft Corporation - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.jdt.ls.core.internal;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import org.eclipse.core.runtime.IConfigurationElement;
+import org.eclipse.core.runtime.ISafeRunnable;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.core.runtime.SafeRunner;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.jdt.ls.core.internal.handlers.BundleUtils;
+import org.eclipse.jdt.ls.core.internal.managers.AbstractProjectsManagerBasedTest;
+import org.junit.Test;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.BundleException;
+
+public class BundleUtilsTest extends AbstractProjectsManagerBasedTest {
+
+	private static final String REFERENCE_PREFIX = "reference:";
+
+	private static final String EXTENSIONPOINT_ID = "testbundle.ext";
+
+	@Test
+	public void testLoadUnload() throws BundleException {
+		Map<String, Object> options = new LinkedHashMap<>();
+		options.put("bundles", new ArrayList<>(Arrays.asList(getBundle())));
+
+		BundleUtils.loadBundles(options);
+
+		String bundleLocation = getBundleLocation(getBundle(), true);
+
+		BundleContext context = JavaLanguageServerPlugin.getBundleContext();
+		Bundle installedBundle = context.getBundle(bundleLocation);
+		assertNotNull(installedBundle);
+		assertEquals(installedBundle.getState(), Bundle.INSTALLED);
+
+		installedBundle.start();

```

Comment:
This shouldn't be necessary. Eclipse runtime should start the bundle when the extension point is instantiated.

Reply:
Well, it seems the bundle is not automatically loaded when it don't implement external extension and it is marked as lazy. If implement external extension, it will start when query the configuration elements.